livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py

```diff
@@ -356,39 +372,48 @@ async def _recv_task() -> None:
                 except Exception:
                     logger.exception("failed to handle event", extra={"event": event})
 
-        input_audio_transcription: session_update_event.SessionInputAudioTranscription | None = None
-        if utils.is_given(self._realtime_model._opts.input_audio_transcription):
-            input_audio_transcription = session_update_event.SessionInputAudioTranscription(
-                model=self._realtime_model._opts.input_audio_transcription.model,
-                language=self._realtime_model._opts.input_audio_transcription.language,
-                prompt=self._realtime_model._opts.input_audio_transcription.prompt,
+        input_audio_transcription = self._realtime_model._opts.input_audio_transcription
+        input_audio_transcription = (
+            session_update_event.SessionInputAudioTranscription.model_validate(
+                input_audio_transcription.model_dump()
+            ).model_dump(
+                by_alias=True,
+                exclude_unset=True,
+                exclude_defaults=True,
             )
+            if input_audio_transcription
+            else None
+        )
 
-        turn_detection: session_update_event.SessionTurnDetection | None = None
-        if utils.is_given(self._realtime_model._opts.turn_detection):
-            turn_detection = session_update_event.SessionTurnDetection.model_validate(
-                asdict(self._realtime_model._opts.turn_detection)
+        turn_detection = self._realtime_model._opts.turn_detection
+        turn_detection = (
+            session_update_event.SessionTurnDetection.model_validate(
+                turn_detection.model_dump()
+            ).model_dump(
+                by_alias=True,
+                exclude_unset=True,
+                exclude_defaults=True,
             )
-
-        session = session_update_event.Session.model_construct(
-            model=self._realtime_model._opts.model,
-            voice=self._realtime_model._opts.voice,
-            input_audio_format="pcm16",
-            output_audio_format="pcm16",
-            modalities=["text", "audio"],
-            turn_detection=turn_detection,
-            input_audio_transcription=input_audio_transcription,
+            if turn_detection
+            else None
```

Comment:
Their server has different structs depending on semantic_vad/server_vad, so you get an error even if the "None" is there. I'm really wondering how they're going to fix their mistake without breaking everyone code.

We exclude the defaults here and not inside the update_session to still allow to disable turn_detection by using None